### PR TITLE
use native Armadillo batch constructor for dgt, dtt and dst type conversion (clean PR)

### PR DESCRIPTION
**Pull request by sgsokol**
Rebase didn't help but a fresh new fork fixed the things.

### CODE REVIEW

File: inst/include/RcppArmadilloAs.h [hunk 1]
```diff
@@ -148,172 +148,20 @@ namespace traits {
...
-                
-                if (diag == "U"){
+            else if (type == "dgTMatrix" || mat.is("dgTMatrix") ||
+                    type == "dtTMatrix" || mat.is("dtTMatrix") ||
+                    type == "dsTMatrix" || mat.is("dsTMatrix")) {
+                arma::urowvec ti = mat.slot("i");
+                arma::urowvec tj = mat.slot("j");
+                arma::Col<T> tx = mat.slot("x");
+                
+                res=arma::sp_mat(true, arma::join_cols(ti, tj), tx, nrow, ncol, true, false);
```
Comment (coatless) [inline inst/include/RcppArmadilloAs.h]: Minor bit... But, could you put a space between `res = arma::sp_mat()`...

**Comment by eddelbuettel** [comment]
Thanks for a cleaner new PR, much appreciated.

Now, @binxiangni made a good point about the code being able to differentiate between different subtype.  Shall we maybe re-add-that feature is a subsequent PR once this one is in?

**Comment by binxiangni** [comment]
A valid but dangerous method to make a clean PR is to `git clone` the `RcppCore/RcppArmadillo` to your local machine, commit on it and then `git push -f` to your forked repo. Then the PR on #174 can be clean if it is not closed. Just be cautious with the repo name in case you overlap your other repo.

**Comment by sgsokol** [comment]
@eddelbuettel a commit 4e3ad83 does this differentiation. I have committed it to my master branch (as instructed by github) but I don't know if it will become a part of this PR or of the next one?

**Comment by sgsokol** [comment]
@binxiangni I have removed my old fork on github, made a new fork and then cloned it locally. It is a little bit radical way but hopefully, it will work without messing things up.

**Comment by eddelbuettel** [comment]
> but I don't know if it will become a part of this PR or of the next one?

If the commit is in the same branch as the previous commits of the PR, then yes.  It just grows as you'd expect.  Will run a new Travis and just appear here.   I'd say try it ...

**Comment by eddelbuettel** [comment]
Actually, it actually is showing two commits:  https://github.com/RcppCore/RcppArmadillo/pull/175/commits/2015884564455d2b9ebd9358de38c1adcb0f4dac and then https://github.com/RcppCore/RcppArmadillo/pull/175/commits/4e3ad83f7bc38cf24fe0616a0f3cabba20b89b6a

**Comment by eddelbuettel** [comment]
This was a *very* important and timely PR.  I had test failures due to `SpMat` changes, but with your syncing it passes.

Now it so happens that I cannot merge this now, but I will give you full credit in `inst/NEWS.Rd`.  Hope that's ok.  

I should have put 8.100.* into a branch earlier.  My bad.

**Comment by eddelbuettel** [comment]
My bad -- similar to Binxiang's patch which I sat on for too long, I now merged this "by hand".  (I neeed Binxiang's for Armadillo 8.100.1, and with that the file was changed.)  

But this too is terrific, and very importamt.  I added both to the ChangeLog and will update NEWS in a moment before I commit.

**Comment by eddelbuettel** [comment]
Oh, and the earlier comment should gave gone to #171. Ooops.

**Comment by eddelbuettel** [comment]
Ok, I folded this into #176 which is now being tested.

**Comment by eddelbuettel** [comment]
The content of this PR is now in master via #176.  Again my apologies for not merging this sooner.

**Pull request closed without merging**